### PR TITLE
chore: make carousel rtl friendly [B1M2-5140]

### PR DESCRIPTION
**Pull request by alexisdevtailor**

### CODE REVIEW

File: packages/components/src/withHorizontalScroll/withHorizontalScroll.ts [hunk 1]
```diff
@@ -92,10 +118,17 @@ const showHideIndicator = (
...
+    lastVisibleIndex < itemRefs.length - 1
+      ? setRightIndicator(true)
+      : setRightIndicator(false);
+    firstVisibleIndex > 0 ? setLeftIndicator(true) : setLeftIndicator(false);
+  } else {
+    lastVisibleIndex < itemRefs.length - 1
+      ? setRightIndicator(true)
+      : setRightIndicator(false);
+    firstVisibleIndex > 0 ? setLeftIndicator(true) : setLeftIndicator(false);
+  }
```
Comment (yarema184) [inline packages/components/src/withHorizontalScroll/withHorizontalScroll.ts:131]: Seems like duplicated code in condition

File: packages/components/src/carousel/control.tsx [hunk 2]
```diff
@@ -35,14 +40,21 @@ export const CarouselControl = styled.button<any>(
     textDecoration: 'none',
     cursor: 'pointer',
     position: 'absolute',
-    left: '-14px',
+    ...(isRtl ? { right: '-14px' } : { left: '-14px' }),
```
Comment (yarema184) [inline packages/components/src/carousel/control.tsx:43]: Can we please use rem

File: packages/components/src/carousel/control.tsx [hunk 3]
```diff
@@ -35,14 +40,21 @@ export const CarouselControl = styled.button<any>(
...
-      left: 'auto',
-      right: '-14px',
+      ...(isRtl
+        ? {
+            left: '-14px',
+            right: 'auto',
+          }
+        : {
+            left: 'auto',
+            right: '-14px',
```
Comment (yarema184) [inline packages/components/src/carousel/control.tsx:56]: Can we please use rem

File: packages/components/src/withHorizontalScroll/withHorizontalScroll.ts [hunk 4]
```diff
@@ -77,12 +81,34 @@ const scrollRightToStep = (
...
+  itemRefs: HTMLElement[],
+  scrollStep: number,
+  toLeft: boolean
+) => {
+  if (toLeft) {
+    const firstVisibleIndex = findFirstVisibleIndex(itemRefs);
+    const actualScrollForIndex =
+      firstVisibleIndex < scrollStep ? 0 : firstVisibleIndex - scrollStep;
+    return actualScrollForIndex;
+  } else {
```
Comment (yarema184) [inline packages/components/src/withHorizontalScroll/withHorizontalScroll.ts:94]: we can don't use else there as we have return in if block. Code will be a little bit clear.

**Review by karl-kallavus — CHANGES_REQUESTED** [review]
Pls fix conflits and comments above

**Pull request merged**